### PR TITLE
212 merge issue

### PR DESCRIPTION
**Pull request by arunraja-hub**
- invoice display
- assay panel

### CODE REVIEW

File: src/actions/game_actions.js [hunk 1]
```diff
@@ -96,6 +106,33 @@ function updateMoney(assays, current_money) {
     dispatch(updateMoneySucceeded(money));
   }; //dispatch money to the store via the updateMoneySucceded synchronous action
 } 
+
+function updateSubTotal(assays, cost_sum) {
+  console.log(assays)
+  // let cost_sum = 0
```
Comment (arunraja-hub) [inline src/actions/game_actions.js:112]: remove this comment

File: src/actions/game_actions.js [hunk 2]
```diff
@@ -96,6 +106,33 @@ function updateMoney(assays, current_money) {
...
+    cost_sum = 0
+    for (const i of assays) {
+      for (const [k, v] of Object.entries(assay_prices)) {
+        if (i === k) {
+          cost_sum += v; //add money for each assay to total cost
+          }
+        }
+      }
+    // console.log('cost_sum in subtotal',cost_sum)
+    // let subtotal =  cost_sum; //calculate new amount of money
```
Comment (arunraja-hub) [inline src/actions/game_actions.js:129]: remove comments

File: src/components/assay/assay.js [hunk 3]
```diff
@@ -9,55 +9,226 @@ import MoleculeImage from "./molecule_image.js";
...
   constructor(props) {
     super(props);
     this.state = {
-      help_toggle: false,
+      invoice_display: false,
+      cost_assays: [],
     };
   }
 
+  // updateSubTotal = () => {
```
Comment (arunraja-hub) [inline src/components/assay/assay.js:28]: remove comment

File: src/components/assay/assay.js [hunk 4]
```diff
@@ -9,55 +9,226 @@ import MoleculeImage from "./molecule_image.js";
...
+      clearance_human: 3.5,
+      logd: 1.5,
+      pampa: 1.0,
+    };
+    console.log(selected_assays)
+    this.props.updateSubTotal(selected_assays,this.props.subtotal)
+    // selected_assays = []
+    
+    
+    // {
```
Comment (arunraja-hub) [inline src/components/assay/assay.js:66]: remove comment

File: src/components/body/footer.js [hunk 5]
```diff
@@ -38,6 +38,7 @@ class Footer extends React.Component {
             </p>
             <p class="m-0 text-end text-white">
               <h5>💰: £{this.props.money}</h5>
+              {/* <h5>Sub-total: £{this.props.subtotal}</h5> */}
```
Comment (arunraja-hub) [inline src/components/body/footer.js:41]: remove comment

**Review by anissa218 — COMMENTED** [review]
Left comments about some code that can be deleted, comments to add, and also to change the naming of show/hide invoice.

File: src/actions/assay_actions.js [hunk 6]
```diff
@@ -96,6 +176,12 @@ function runAssay(selected_mol, assays) {
   };
 }
 
+function calcAssay(selected_mol, assays) {
```
[the remark this answers is not in the document]
Comment (arunraja-hub) [reply src/actions/assay_actions.js:179]: yep deleted

File: src/components/assay/assay.js [hunk 7]
```diff
@@ -9,55 +9,226 @@ import MoleculeImage from "./molecule_image.js";
...
+  //         this.updateSubTotal();
+  //       }
+  //     }
+  //   }
+  //   // assays_run["drug_props"] = true;
+  //   this.resetCostAssays();
+  //   this.props.calcAssay(this.props.selected_mol, toggle_assay);
+  // };
+  
+  onShow = (event) => {
```
[the remark this answers is not in the document]
Comment (arunraja-hub) [reply src/components/assay/assay.js:132]: deleted

File: src/components/assay/assay.js [hunk 8]
```diff
@@ -9,55 +9,226 @@ import MoleculeImage from "./molecule_image.js";
...
+                      </table> 
+
+                    </div>
+                  )}
+                </div>
+              )}
+              {this.props.invoice_display == false && (
+                <div className="invoice-inactivebutton">
+                  <button onClick={() => {this.invoiceDisplay(); this.showInvoice();this.calcAssays();}}>
+                    show invoice
```
[the remark this answers is not in the document]
Comment (arunraja-hub) [reply src/components/assay/assay.js:217]: changed to update invoice with comment to double click to view latest sub-total

**Review by anissa218 — APPROVED** [review]

**Pull request merged**